NDTensors/src/dense/fill.jl

```diff
@@ -22,18 +22,27 @@ function generic_randn(StoreT::Type{<:Dense}, dim::Integer=0)
 end
 
 function generic_zeros(
-  StoreT::Type{<:Dense{ElT,DataT}}, dim::Integer=0
+  StoreT::Type{<:Dense{ElT,DataT}}, dim::Integer
 ) where {DataT<:AbstractArray,ElT}
   @assert ElT == eltype(DataT)
   data = generic_zeros(DataT, dim)
   StoreT = set_datatype(StoreT, typeof(data))
   return StoreT(data)
 end
 
-function generic_zeros(StoreT::Type{<:Dense{ElT}}, dim::Integer=0) where {ElT}
+function generic_zeros(
+  StoreT::Type{<:Dense{ElT,DataT}}, dims
+) where {DataT<:AbstractArray,ElT}
+  @assert ElT == eltype(DataT)
+  data = generic_zeros(DataT, dim(dims))
+  StoreT = set_datatype(StoreT, typeof(data))
+  return StoreT(data)
+end
+
+function generic_zeros(StoreT::Type{<:Dense{ElT}}, dim) where {ElT}
   return generic_zeros(default_storagetype(ElT), dim)
 end
 
-function generic_zeros(StoreT::Type{<:Dense}, dim::Integer=0)
+function generic_zeros(StoreT::Type{<:Dense}, dim)
   return generic_zeros(default_storagetype(), dim)
 end
```

Comment:
Same with this one, seems like this should specify default values of `eltype` and `datatype` of `StoreT`. This seems to implicitly rely on the fact that `default_storagetype()` makes a `Dense`, but I don't think we should rely on that behavior in code like this.

Reply:
Okay I removed this code. thanks!